llvm/include/llvm/ExecutionEngine/Orc/TargetProcess/LibraryScanner.h

```diff
@@ -338,14 +424,14 @@ class LibraryScanHelper {
   addBasePath(const std::string &P,
               PathType Kind =
                   PathType::Unknown); // Add a canonical directory for scanning
-  std::vector<std::shared_ptr<LibrarySearchPath>>
-  getNextBatch(PathType Kind, size_t batchSize);
+  std::vector<const LibrarySearchPath *> getNextBatch(PathType Kind,
```

Comment:
I think we should be careful with such return types, let's have them as out parameters so make sure things are consistently optimized. 

Please also update such things elsewhere in the PR.